goal_src/jak2/engine/gfx/ocean/ocean.gc

```diff
@@ -5,5 +5,1374 @@
+              )
+          )
+         ((begin
+            (let ((a0-8 (logand (the int (* 0.000010172526 f30-0)) 3))
+                  (a3-4 (logand (the int (* 0.000010172526 f28-0)) 3))
+                  (v1-10 (-> obj ocean-near-indices data (-> v1-7 child)))
+                  )
+              (set! v1-12 (-> (the-as (pointer int16) (+ (* (+ (* a3-4 4) a0-8) 2) (the-as int v1-10)))))
+              )
+            (= v1-12 -1)
+            )
+          (if arg1
+              (-> obj start-corner y)
+              4095996000.0
+              )
+          )
+         (else
+           (let ((a0-14 (logand (the int (* 0.00008138021 f30-0)) 7)))
+             (cond
+               ((not (logtest? (-> obj ocean-mid-masks data v1-12 mask (logand (the int (* 0.00008138021 f28-0)) 7)) (ash 1 a0-14))
+                     )
+                (let* ((f1-2 (vector-vector-distance arg0 (math-camera-pos)))
+                       (f26-0 (- 1.0 (fmin 1.0 (* 0.000010172526 f1-2))))
+                       )
+                  (if (-> obj ocean-near-translucent?)
+                      (+ (* f26-0 (set-corners! obj f30-0 f28-0)) (-> obj start-corner y))
+                      (-> obj start-corner y)
+                      )
+                  )
+                )
+               (arg1
+                 (-> obj start-corner y)
+                 )
+               (else
+                 4095996000.0
+                 )
+               )
+             )
+           )
+         )
+       )
+     )
+    (arg1
+      0.0
+      )
+    (else
+      4095996000.0
+      )
+    )
+  )
+
+(defun init-ocean-far-regs ()
+  (local-vars (v1-3 float) (a0-4 float) (a0-5 float))
+  (rlet ((vf13 :class vf)
+         (vf14 :class vf)
+         (vf25 :class vf)
+         (vf26 :class vf)
+         (vf27 :class vf)
+         (vf28 :class vf)
+         (vf29 :class vf)
+         (vf30 :class vf)
+         (vf31 :class vf)
+         )
+    (let ((v1-0 *math-camera*))
+      (set-vector! (-> *sky-work* fog) (-> v1-0 pfog0) (-> v1-0 fog-min) (-> v1-0 fog-max) 3071.0)
+      (cond
+        ((-> *time-of-day-context* use-camera-other)
+         (.lvf vf31 (&-> v1-0 camera-temp-other vector 0 quad))
+         (.lvf vf30 (&-> v1-0 camera-temp-other vector 1 quad))
+         (.lvf vf29 (&-> v1-0 camera-temp-other vector 2 quad))
+         (.lvf vf28 (&-> v1-0 camera-temp-other trans quad))
+         (.mov a0-4 vf28)
+         )
+        (else
+          (.lvf vf31 (&-> v1-0 camera-temp vector 0 quad))
+          (.lvf vf30 (&-> v1-0 camera-temp vector 1 quad))
+          (.lvf vf29 (&-> v1-0 camera-temp vector 2 quad))
+          (.lvf vf28 (&-> v1-0 camera-temp trans quad))
+          (.mov a0-5 vf28)
+          )
+        )
+      (.lvf vf26 (&-> v1-0 inv-hmge-scale quad))
+      (.lvf vf14 (&-> v1-0 hmge-scale quad))
+      (.lvf vf25 (&-> v1-0 hvdf-off quad))
+      )
+    (.lvf vf13 (&-> *sky-work* fog quad))
+    (.lvf vf27 (&-> *sky-work* giftag-ocean quad))
```

Comment:
This function stores stuff in vf13 and vf27, to be used by the large polygon stuff. In OpenGOAL, we don't save the registers (x86 doesn't have enough), so we have to do some rewriting.

You can look at jak 1's version of this as an example. You can convert this to mips2c, then put
```
ocean_regs_vfs.copy_vfs_from_other(c);
```
at the very bottom.

Then, at the top of `render_ocean_quad::execute`, you put
```
c->copy_vfs_from_other(&ocean_regs_vfs);
```
and this will make sure that the large polygon drawing code has the values for vf13 and vf27 set here.